src/main/rx/rx_spi_common.c

```diff
@@ -133,6 +136,7 @@ bool rxSpiCheckBindRequested(bool reset)
         return false;
     } else {
         if (reset) {
+            rxSpiNormalSpeed();
```

Comment:
I think this should go into the `if` case above - this is where we don't bind.

Reply:
With my commit as is, see below.

![image](https://user-images.githubusercontent.com/11480839/126778384-c88941db-fb64-4086-afa0-d3908c39d046.png)

We get one call to `rxSpiBindSpeed()`, and then one to `rxSpiNormalSpeed()` once binding is complete. There is then a period before I take the tx out of bind mode. The prior `!bindRequested` condition is being hit whilst not in bind repeatedly.

Reply:
Interesting.
Now to my reading of https://github.com/betaflight/betaflight/blob/1552e30d57a400a9da1bfd835e637a263d701f5f/src/main/rx/cc2500_frsky_shared.c#L371-L381, what your commit does is resetting SPI to full speed _at the beginning_ of the bind. But since this makes binding work, I suspect that we have been looking at this the wrong way - it is not the binding itself where full speed SPI is problematic, but the CC2500 initialisation that precedes the binding. This kind of makes sense, if there are any hardware settings that are changed that result in the chip blocking for some time to apply the change.
So I think something like this will make it a bit more obvious what actually is done:
```
diff --git a/src/main/drivers/rx/rx_spi.c b/src/main/drivers/rx/rx_spi.c
index 1bdbbef44..98c42577e 100644
--- a/src/main/drivers/rx/rx_spi.c
+++ b/src/main/drivers/rx/rx_spi.c
@@ -45,8 +45,8 @@
 
 // 13.5 MHz max SPI frequency
 #define RX_MAX_SPI_CLK_HZ 13500000
-// 6.5 MHz max SPI frequency during binding
-#define RX_BIND_MAX_SPI_CLK_HZ 6500000
+// 6.5 MHz max SPI frequency during startup
+#define RX_STARTUP_MAX_SPI_CLK_HZ 6500000
 
 static extDevice_t rxSpiDevice;
 static extDevice_t *dev = &rxSpiDevice;
@@ -80,9 +80,9 @@ void rxSpiNormalSpeed()
     spiSetClkDivisor(dev, spiCalculateDivider(RX_MAX_SPI_CLK_HZ));
 }
 
-void rxSpiBindSpeed()
+void rxSpiStartupSpeed()
 {
-    spiSetClkDivisor(dev, spiCalculateDivider(RX_BIND_MAX_SPI_CLK_HZ));
+    spiSetClkDivisor(dev, spiCalculateDivider(RX_STARTUP_MAX_SPI_CLK_HZ));
 }
 
 bool rxSpiDeviceInit(const rxSpiConfig_t *rxSpiConfig)
diff --git a/src/main/drivers/rx/rx_spi.h b/src/main/drivers/rx/rx_spi.h
index 3580452e4..2633bcc93 100644
--- a/src/main/drivers/rx/rx_spi.h
+++ b/src/main/drivers/rx/rx_spi.h
@@ -33,7 +33,7 @@ struct rxSpiConfig_s;
 void rxSpiDevicePreInit(const struct rxSpiConfig_s *rxSpiConfig);
 bool rxSpiDeviceInit(const struct rxSpiConfig_s *rxSpiConfig);
 void rxSpiNormalSpeed();
-void rxSpiBindSpeed();
+void rxSpiStartupSpeed();
 void rxSpiDmaEnable(bool enable);
 uint8_t rxSpiTransferByte(uint8_t data);
 void rxSpiWriteByte(uint8_t data);
diff --git a/src/main/rx/cc2500_frsky_shared.c b/src/main/rx/cc2500_frsky_shared.c
index adee361a4..1fa8e2f39 100644
--- a/src/main/rx/cc2500_frsky_shared.c
+++ b/src/main/rx/cc2500_frsky_shared.c
@@ -155,6 +155,8 @@ const cc2500RegisterConfigElement_t cc2500FrskyXLbtV2Config[] =
 };
 
 static void initialise() {
+    rxSpiStartupSpeed();
+
     cc2500Reset();
 
     cc2500ApplyRegisterConfig(cc2500FrskyBaseConfig, sizeof(cc2500FrskyBaseConfig));
@@ -197,6 +199,8 @@ static void initialise() {
         calData[c][1] = cc2500ReadReg(CC2500_24_FSCAL2);
         calData[c][2] = cc2500ReadReg(CC2500_25_FSCAL1);
     }
+
+    rxSpiNormalSpeed();
 }
 
 void initialiseData(bool inBindState)
diff --git a/src/main/rx/rx_spi_common.c b/src/main/rx/rx_spi_common.c
index 7802b4608..105bca3ac 100644
--- a/src/main/rx/rx_spi_common.c
+++ b/src/main/rx/rx_spi_common.c
@@ -117,8 +117,6 @@ void rxSpiLedBlinkBind(void)
 
 void rxSpiBind(void)
 {
-    rxSpiBindSpeed();
-
     bindRequested = true;
 }
 
@@ -136,7 +134,6 @@ bool rxSpiCheckBindRequested(bool reset)
         return false;
     } else {
         if (reset) {
-            rxSpiNormalSpeed();
             bindRequested = false;
         }
```

Reply:
@mikeller Please see latest commit. Still working for me with this change. I think your rx is more sensitive to binding for some reason. Thanks for looking into this.

Reply:
This stuff is so complicated!  Thank you for getting so deeply into it!